src/NuGet.Core/NuGet.Commands/SignCommand/SignCommandRunner.cs

```diff
@@ -115,6 +125,24 @@ private int SignPackage(string packagePath, string outputPath, ILogger logger, I
             return 0;
         }
 
+        private static async Task AddSignatureAsync(ILogger logger, ISignatureProvider signatureProvider, SignPackageRequest request, FileStream packageWriteStream)
```

Comment:
These methods should pass a cancellation token down. It is fine if None is passed from nuget.exe, but a token should be supported.